### ISSUE TITLE
Solution to improve complicated ACRE and ORBAT settings

### ISSUE THREAD

**Issue by BorderKeeper** [TMF3/TMF#311]
Hi, as a mission maker from ARCOMM where we adopted TMF recently I have came across TMF_AcreSettings and TMF_OrbatSettings a couple times both from technical (debugging the mission.sqm) and mission maker (using the TMF setting windows) perspectives.
Now the reason why I am creating this ticket is because most of my experiences with those two have been negative and I can hear a similar story from other ARCOMM mission makers.

**Problems**:
- The ACRE radio assignment window is quite tough to work with due to it no supporting mass-editing and collapsing/expanding the right tree on every click. There is also **a lot** of tedious error-prone work when working with larger ORBATs, which is a necessity for a mission with 40-60 people present.
- Since both ORBAT and ACRE settings seem to be bound to player/group entities creating or removing ORBAT units is a tedious job, where sometimes I just have my radios randomly get reassigned.
- The ORBAT settings manager suffers from similar UX issues as the ACRE one
- The automatic radio assignment is great, but it doesn't scale with big ORBATs (not talking about TMF not squashing 343 nets so Foxtrot does not end up on Block 2, that is a different problem)

**ORBAT Solution**:
- Rather than constructing hierarchies in the TMF settings, make an option to select a Parent Group in the Group Marker and Specialist Group Marker Tabs. That way even if I decide to create A4 I will not have to also add it in the TMF settings, but know it's parent group (Alpha Squad) has been carried over. The user will still have to create the "Parent" group fist, but it will greatly reduce the time we will have to spend there.
![TMFOrbatFix](https://user-images.githubusercontent.com/7063238/83943684-65393080-a7fe-11ea-9ed2-5c30186efabe.png)

**ACRE Solution**:
- Remove the auto-assignment of nets and leave it optional for 343 short range net only.
- Remove network configuration all-together
- Create a new tab under Specialist Unit Marker for each unit where you will be able to assign radios for the person
- Create a check mark in ACRE2 Settings that will control automatic 343 net assignment
- The net 343 assignment will be done using groups, where each group will get a net and all people with a 343 in that group will be set to that net.
- Above will free us from working with the Network manager with the issue of leads having to set their radio 148 and 152 nets (which seems like a much better alternative)
![TMFAcreFix](https://user-images.githubusercontent.com/7063238/83943851-79316200-a7ff-11ea-93ce-f0c9e8283f90.png)

Cheers, BorderKeeper.

**Comment by AVESUM** [comment]
Our group find current Network Configurator great and simple for 148/152/117. Named channels are great! Of course we use a framework mission as template so that everything is set as default when starting mission creation in Eden. 

Two confusing things are
1) the conflict between Network assignment of radios and the loadout assignment of radios. We removed the latter completely and rely on the Network Configuration.
2) lack of control in channels and blocks of the 343, it's problematic in-game to know which channel/block to switch to when the need arises to contact another unit. We utilize approximately 20 different squad nets on the 343.

**Comment by Sniperhid** [comment]
Hi BorderKeeper,

Firstly thanks for feedback, I don't think in the last 4 years of the ORBAT/ACRE settings been there I have anyone write anything close to this much on them. I also want to say I admire someone who comes with ideas for solutions and not just list problems.

Now let's talk about the problems. I am actually curious to hear more about these and examples. Let's focus on the ACRE2 radio allocation. I do acknowledge the UI is not the best, unfortunately Arma is a bit crude in what it offers.

The goal of these editing 'pages' where to give a high level overview of both radio nets/channels whilst offering the ability to edit them. In additional a key design goal was something that could be flexible enough to support virtually any radio net allocation for any size - I wanted these systems to be able to fit any community requirement. We have comfortably used it for company level operations with more than 70 players.

To address your point on squashing 343 nets. What do you mean? The system allows you to manually handpick who is given a radio for every channel, so as I see it this falls all on what the mission making sets up in interface. Any non-random radio allocation from another system should be reproducible here.

**Comment by BorderKeeper** [comment]
@Sniperhid after a discussion with @Freddo3000 and taking a good look at how the data inside mission.sqm get's represented I see that you are mostly already doing what I described so let me rephrase my post above a bit.

Radio and ORBAT placement is saved under a specific group, it is only represented as one big list under attributes edit windows. I have falsely though that not only you store the placement in the big ORBATSettings/ACRESettings arrays in missionNamespace, but that you also tie the groups using EntityIds.

This fixes a lot, but there is still the issue of radios, right now if I want to create a new fireteam (A4 under ASL for example) I have to:
- Copy and paste A3
- Rename Callsign, Group marker and Slotting name
- Since group attributes got carried over A4 already is under ASL (but should probably be last in the list so thanks to my OCD will have to click 5 times to move it at the bottom of Alpha Squad)
![image](https://user-images.githubusercontent.com/7063238/83968164-874ab580-a8c7-11ea-8018-681f1160ec7c.png)
- Go to ACRE settings and remove A4 from being on A3 net, create A4 net and move them onto it.
![radio_nets](https://user-images.githubusercontent.com/7063238/83968275-40a98b00-a8c8-11ea-81a2-c686f3183bf0.gif)

Now this is all fine, except for the ACRE part which is way too time consuming compared to the other steps. I would take the name from the group name of the unit and make 343 assignment automatic (or at least have 343 channel settings under group attributes not in missionNamespace so it gets copy pasted with the group)
To add to above, we have an overly complex ORBAT (2PLTs) that we can slim down a lot to get the net counts down, but since you mentioned you worked with it with 70 people then it should be a good example still.

**Net squashing:**
As you can see above with a large ORBAT there are a lot of 343 nets. Thanks to how time consuming the ACRE editation window is mission makers will not remove nets they are not planning to use (IE. 4 Gambler groups or the entirety of 2PLT 343 nets).
This means that if they choose to use for example 2PLT Foxtrot 1, that group will be on Block 2 Channel 10, even though we only have 7 343 nets in total.

At the start I proposed rather than assigning specific Channels/Blocks to a net, the system would dynamically assign these incrementally on mission start according to the groups that actually are playing that op. Alternative to this fix would be what I proposed above, ie. to move/add 343 channel assignment to group settings (same as you do with ORBAT marker assignments). This means that if the group is deleted it's net is deleted with it and it does not inflate the net count artifically.

Cheers, BorderKeeper.

**Comment by Sniperhid** [comment]
So admittedly generally in the operations I play we utlize a 343 per squad rather than fireteam which results in significantly less radio nets (pretty much always single block). Our use case is typically having a 343 span multiple groups.

I can see what your saying around the usefulness of having a 343 net potentially be a group attribute. As this best fits your radio nets. This is where there is a design trade off to be made. As you say there is the potential for it to be a group attribute but this then fits the attribute to the group and then negatively impacts 343 nets that need to span multiple groups aka my communities' usecase. Even though the 'master list' is more time consuming it still allows both approaches but the trade off is that it is indeed more time consuming in your case. To this end the TMF philsophy here would be to preserve flexability.

Filtering the created radio channels to present groups/units does make more sense, but isn't without its challenges. This does rely on at least one unit from group existing at mission start. There are some edge cases, at present radio nets typically exist at briefing time and if you are briefing on the map screen and a player JIPs into a group that didn't previously exist which has its own nets this then cuases an Issue. To correct this either the presets would need to recreated or the radio channel creation delayed effectively denying the automatically generated briefing page during the briefing (and the comms plan is useful). There is also the edge case that a mission maker may want a channel to exist and show up even if no one is allocated it.

The best trick I can currently offer to lower time spent in those menus is to have templates with pre-existing ORBAT so the cost is spent once.

I should clarify that I think there is room for improvement here such as:
- Displaying Block number on briefing page for 343 channels (if more than 16 channels)
- Updating documentation to clarify what happens with loadout gear and network assignment (It attempts to use radios issued with loadout gear to full radio requirements first).
- Some ability to influence block selection (This is perhaps where being able to use the ORBAT heirarchy might help - but I imagine an extremely complex change)
- Ability to reorder radio channels
